### PR TITLE
98 eg path length global

### PR DESCRIPTION
**Pull request by egolol**
The code is incomplete but I need review and answer to what is the average whitin subgraphs

### CODE REVIEW

File: braph/graph/measures/PathLengthGlobal.m [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                pathLength = g.getMeasureValue('PathLength');
+            else
+                pathLength = calculate@PathLength(m);
+            end
+            
+            settings = m.getSettings();
+            pathLengthGlobal_rule = get_from_varargin(0, 'AveragePathLengthRule', settings{:});
+            switch lower(pathLengthGlobal_rule)
+                case {'subgraphs'}  % whitin connected subgraphs
+
```
Comment (egolol) [inline braph/graph/measures/PathLengthGlobal.m]: What should this look like?

File: braph/graph/test_Measure.m [hunk 2]
```diff
@@ -17,7 +17,7 @@
                 ['BRAPH:' measure_class ':getValueBroken'], ...
                 [measure_class '.getValue()  used with ' graph_class ' should return a scalar'])
         end
-        if m.is_nodal()
+        if m.is_nodal() & ~isequal(measure_class, 'PathLength')
```
Comment (giovannivolpe) [inline braph/graph/test_Measure.m]: This seems odd. Why?

Also, path length should give this format...

File: braph/graph/measures/test_PathLength.m [hunk 3]
```diff
@@ -0,0 +1,59 @@
+% test PathLength
```
Comment (giovannivolpe) [inline braph/graph/measures/test_PathLength.m]: Add also a file for test_PathLengthAv

File: braph/graph/measures/PathLength.m [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    methods
+        function m = PathLength(g, varargin)
+            m = m@Measure(g, varargin{:});
+        end
+    end
+    methods (Access = protected)
+        function P =  calculate(m)
+            g = m.getGraph();
+            D = g.getMeasure('Distance').getValue();
+            D = dediagonalize(D);
```
Comment (giovannivolpe) [inline braph/graph/measures/PathLength.m]: D is already with 0 on diagonal

File: braph/graph/measures/PathLength.m [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            
+            m.pathLengthIn = zeros(1, N);
+            
+            in_cond1 = D(:,1)==Inf;  % locates all not connected nodes           
+            D(in_cond1,:) = [];  % removes conditions          
+            m.pathLengthIn= mean(D,1);
+            
+            out_cond1 = Dout(1, :) == Inf;  % locates all not connected nodes      
+            Dout(:,out_cond1 ) = [];  % removes conditions
+            m.pathLengthOut =mean(Dout,2)';
```
Comment (giovannivolpe) [inline braph/graph/measures/PathLength.m]: PathLength should only be defined for BU and WU

For D graph, let's have InPathLength and OutPathLength

File: braph/graph/measures/PathLength.m [hunk 6]
```diff
@@ -0,0 +1,67 @@
+classdef PathLength < Measure
+    properties
+        pathLengthOut
+        pathLengthIn
+    end
```
Comment (giovannivolpe) [inline braph/graph/measures/PathLength.m]: Not needed, see below.

File: braph/graph/measures/PathLengthGlobal.m [hunk 7]
```diff
@@ -0,0 +1,70 @@
+classdef PathLengthGlobal < PathLength
```
Comment (giovannivolpe) [inline braph/graph/measures/PathLengthGlobal.m]: Call it PathLengthAv for consistency

File: braph/graph/measures/PathLength.m [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    methods
+        function m = PathLength(g, varargin)
+            m = m@Measure(g, varargin{:});
+        end
+    end
+    methods (Access = protected)
+        function P =  calculate(m)
+            g = m.getGraph();
+            D = g.getMeasure('Distance').getValue();
```
Comment (giovannivolpe) [inline braph/graph/measures/PathLength.m]: Let's calculate is as

           if g.is_measure_calculated('Triangles')
                triangles = g.getMeasureValue('Triangles');
            else
                triangles = calculate@Triangles(m);
            end

Comment (egolol) [reply braph/graph/measures/PathLength.m]: But Distance is not a base class of PathLength

**Comment by egolol** [comment]
Im going to erase this pull request, and restart

**Pull request closed without merging**